tests/pipelines/test_api.py

```diff
@@ -192,3 +143,74 @@ def test_delete(get_credentials_mock, pipelines_api):
     assert delete_mock.call_args[0][0] == PIPELINE_ID
     assert delete_mock.call_args[0][1] == CREDENTIALS
     assert delete_mock.call_args[0][2] == HEADERS
+
+
+def test_partition_local_remote(pipelines_api):
+    libraries = [
+        LibraryObject('jar', '/absolute/path/abc.ext'),
+        LibraryObject('jar', 'relative/path.ext'),
+        LibraryObject('jar', 'file:///file/scheme/abs/path.ext'),
+        LibraryObject('jar', 'file://file/scheme/two/slashes/path.ext'),
+        LibraryObject('jar', 'FILE:/all/caps.ext'),
+        LibraryObject('jar', 'FiLe:/weird/case.ext'),
+        LibraryObject('jar', 'file.ext'),
+        # remote
+        LibraryObject('jar', 'dbfs:/absolute/path/abc.ext'),
+        LibraryObject('jar', 's3:file:/file/scheme/abs/path.ext'),
+        LibraryObject('jar', 'scheme:file.ext'),
+        LibraryObject('jar', 'scheme:/abs/path.ext'),
+        LibraryObject('jar', 'scheme://abs/path.ext'),
+        LibraryObject('egg', 'file:/abs/path.ext'),
+        LibraryObject('whl', 'rel/path.ext')
+    ]
+    expected_llo = [
+        LibraryObject('jar', '/absolute/path/abc.ext'),
+        LibraryObject('jar', 'relative/path.ext'),
+        LibraryObject('jar', '///file/scheme/abs/path.ext'),
+        LibraryObject('jar', '//file/scheme/two/slashes/path.ext'),
+        LibraryObject('jar', '/all/caps.ext'),
+        LibraryObject('jar', '/weird/case.ext'),
+        LibraryObject('jar', 'file.ext')
+    ]
+    expected_rest = [
+        LibraryObject('jar', 'dbfs:/absolute/path/abc.ext'),
+        LibraryObject('jar', 's3:file:/file/scheme/abs/path.ext'),
+        LibraryObject('jar', 'scheme:file.ext'),
+        LibraryObject('jar', 'scheme:/abs/path.ext'),
+        LibraryObject('jar', 'scheme://abs/path.ext'),
+        LibraryObject('egg', 'file:/abs/path.ext'),
+        LibraryObject('whl', 'rel/path.ext')
+    ]
+    llo, rest = pipelines_api._partition_libraries_and_extract_local_paths(libraries)
+    assert llo == expected_llo
+    assert rest == expected_rest
```

Comment:
just for completeness, let's make sure that expected_llo + expected_rest == libraries

Reply:
That's not true because the function also strips out the file: prefix.